internal/node/hybrid/validator.go

```diff
@@ -107,20 +108,66 @@ func validateRolesAnywhereNode(node *api.NodeConfig) error {
 	if len(node.Spec.Hybrid.IAMRolesAnywhere.NodeName) > 64 {
 		return fmt.Errorf("NodeName can't be longer than 64 characters in hybrid iam roles anywhere configuration")
 	}
+
+	// IAM roles anywhere certificate validation
 	if node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath == "" {
 		return fmt.Errorf("CertificatePath is missing in hybrid iam roles anywhere configuration")
 	}
-	if node.Spec.Hybrid.IAMRolesAnywhere.PrivateKeyPath == "" {
-		return fmt.Errorf("PrivateKeyPath is missing in hybrid iam roles anywhere configuration")
-	}
-
 	if !file.Exists(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath) {
 		return fmt.Errorf("IAM Roles Anywhere certificate %s not found", node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath)
 	}
+	if err := ValidateCertificate(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath, nil); err != nil {
+		var validationErr *ValidationError
+		if errors.As(err, &validationErr) {
+			updateIAMRAErrorMessage(node.Spec.Hybrid.IAMRolesAnywhere.CertificatePath, validationErr)
+		}
+		return fmt.Errorf("IAM Roles Anywhere certificate error - %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("validating IAM Roles Anywhere certificate: %w", err)
```

this would be the idiomatic way in go